cmd/juju/cloud/list.go

```diff
@@ -242,10 +278,34 @@ func (c *cloudList) all() map[string]*CloudDetails {
 	return result
 }
 
-func listCloudDetails(store jujuclient.CredentialGetter) (*cloudList, error) {
+func (c *cloudList) filter(all bool) bool {
+	if all {
+		return false
+	}
+	if len(c.personal) == 0 && len(c.builtin) == 0 && len(c.public) == 0 && len(c.remote) == 0 {
+		return false
+	}
+
+	result := false
+	examine := func(someClouds map[string]*CloudDetails) {
+		for name, cloud := range someClouds {
+			if cloud.CredentialCount == 0 {
```

Comment:
Will this break the display of LXD and microk8s where there's no credential up front but we create one on the fly the first time.
This is also a breaking CLI change. Is this what we want?

Reply:
Yes, this has been insisted on by both @jameinel and @mitechie as the most useful information that this command can produce. We have all agreed.

Reply:
You are right. I totally forgot that had been discussed.